tutorials/appengine-serve-machine-learning-model/index.md

```diff
@@ -0,0 +1,176 @@
+```
+
+We store the model as a pickled file on [Cloud Storage][storage].  To make sure the App Engine service knows where to find the model, we pass in the model's bucket and file name as environment variables.  This is done in the `app.yaml` configuration file.
+
+
+### [`app.yaml`][app.yaml]
+
+The `app.yaml` configuration file defines the App Engine service.  We define environment variables pointing to the trained model:
+
+```yaml
+env_variables:
+    # The app will look for the model file at: gs://MODEL_BUCKET/MODEL_FILENAME
+    MODEL_BUCKET: BUCKET_NAME
+    MODEL_FILENAME: lr.pkl
+```
+
+You should replace `BUCKET_NAME` with a bucket owned by your project.  `lr.pkl` is a simple linear regression model included in the sample app.
+
+To use Cloud Endpoints to manage the service, we need to specify a `config_id` under the `endpoints_api_service` field:
+
+```yaml
+endpoints_api_service:
+  name: modelserve-dot-PROJECT_ID.appspot.com
+  config_id: CONFIG_ID
+```
+
+The `CONFIG_ID` points to a Cloud Endpoints service deployment.  You can find all the deployments on the [Cloud Endpoints console][endpoints] under each service page's Deployment history tab.
+
+To deploy a service to Cloud Endpoints, we configure it with the `modelserve.yaml` file.
+
+### [`modelserve.yaml`][modelserve.yaml]
+
+The `modelserve.yaml` configuration file defines the service according to the [OpenAPI specification][openapi].  We highlight only some of the key configurations below.
+
+- Specify the host:
+
+  ```yaml
+  host: "modelserve-dot-PROJECT_ID.appspot.com"
+  ```
+
+  This host means we will handle the requests with an App Engine service called `modelserve`.
+
+- Enforce authentication with API key by adding the `security` and `securityDefinitions` fields:
+
+    ```yaml
+    security:
+      - api_key: []
+    securityDefinitions:
+      api_key:
+        type: "apiKey"
+        name: "key"
+        in: "query"
+    ```
+
+    This is optional, but allows you to grant service consumer permissions on the [Cloud Endpoints console][endpoints].  The API key must be associated to a Google Cloud Platform project.  You can create API keys on the [credentials][credentials] page.
+
+- Additionally, you can configure quota for each consumer at the project level.  First we specify a service level metric in order to track the number of requests:
+
+  ```yaml
+  x-google-management:
+    metrics:
+      - name: "modelserve-predict"
+        displayName: "modelserve predict"
+        valueType: INT64
+        metricKind: DELTA
+    quota:
+      limits:
+        - name: "modelserve-predict-limit"
+          metric: "modelserve-predict"
+          unit: "1/min/{project}"
+          values:
+            STANDARD: 1000
+  ```
+
+  This configurations declares a metric `modelserve-predict` and sets its limit to 1000 units per minutes per project.
+
+  To specify requests to which paths will be counted towards this metric, we add the following in the `paths` field:
```

Comment:
Only requests to specified paths are counted towards this metric. Specify these paths by adding the following in the `paths` field: